internal/services/notification/runwebhookdeliverys.go

```diff
@@ -0,0 +1,215 @@
+
+	err := n.d.Do(ctx, func(tx *sql.Tx) error {
+		// start from last event
+		runWebhookDelivery, err := n.d.GetLastRunWebhookDelivery(tx)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if runWebhookDelivery == nil {
+			return nil
+		}
+
+		curSequence = runWebhookDelivery.Sequence
+
+		return nil
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	for {
+		runWebhookDeliveryIDs := make([]string, 0)
+		err := n.d.Do(ctx, func(tx *sql.Tx) error {
+			var err error
+			runWebhookDeliverys, err := n.d.GetRunWebhookDeliverysFromSequence(tx, curSequence, MaxRunWebhookDeliverysLimit)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			for _, r := range runWebhookDeliverys {
+				runWebhookDeliveryIDs = append(runWebhookDeliveryIDs, r.ID)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		for _, id := range runWebhookDeliveryIDs {
+			sequence, err := n.fetchRunWebhookDelivery(ctx, id)
+			if err != nil {
+				n.log.Error().Msgf("failed to trigger run webhook delivery event")
+			}
+			if sequence != 0 {
+				curSequence = sequence
+			}
+		}
+	}
+}
+
+func (n *NotificationService) fetchRunWebhookDelivery(ctx context.Context, runWebhookDeliveryID string) (uint64, error) {
+	l := n.lf.NewLock(fmt.Sprintf("%s-%s", RunWebhookDeliverysLockKey, runWebhookDeliveryID))
+	if err := l.TryLock(ctx); err != nil {
+		if errors.Is(err, lock.ErrLocked) {
+			return 0, nil
+		}
+		return 0, errors.WithStack(err)
+	}
+	defer func() {
+		_ = l.Unlock()
+	}()
+
+	var runWebhook *types.RunWebhook
+	var runWebhookDelivery *types.RunWebhookDelivery
+
+	var sequence uint64
+	err := n.d.Do(ctx, func(tx *sql.Tx) error {
+		var err error
+
+		runWebhookDelivery, err = n.d.GetRunWebhookDeliveryByID(tx, runWebhookDeliveryID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		sequence = runWebhookDelivery.Sequence
+
+		runWebhook, err = n.d.GetRunWebhookByID(tx, runWebhookDelivery.RunWebhookID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return sequence, errors.WithStack(err)
+	}
+
+	if runWebhookDelivery.DeliveryStatus == types.NotDelivered {
+		err = n.d.Do(ctx, func(tx *sql.Tx) error {
+			var err error
+
+			runWebhookDelivery, err = n.d.GetRunWebhookDeliveryByID(tx, runWebhookDeliveryID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return sequence, errors.WithStack(err)
+		}
+
+		var webhook RunWebhook
+		switch runWebhook.PayloadVersion {
+		case 1:
+			if err = json.Unmarshal([]byte(runWebhook.Payload), &webhook); err != nil {
+				return sequence, errors.WithStack(err)
+			}
+		default:
+			return sequence, errors.Errorf("unknown payload version: %d", runWebhook.PayloadVersion)
+		}
+
+		webhookDelivered := false
+		resp, deliveryDuration, err := n.sendRunWebhook(ctx, &webhook, runWebhook.UUID)
+		if err == nil && resp != nil && resp.StatusCode == http.StatusOK {
```

Comment:
Add a note that err != nil is not checked because every error is considered a failed delivery.